src/components/Functions/dataComponentMapper.tsx

```diff
@@ -0,0 +1,22 @@
+import { ExclamationCircleIcon as ExclamationCircleIcon } from '@patternfly/react-icons';
+import React from 'react';
+
+export const dataComponentMapper = (name: string): React.ReactNode => {
+  switch (name) {
+    case 'ExclamationCircleIcon':
+      return <ExclamationCircleIconPoint />;
+      break;
+    default:
+      return null;
+  }
+};
+
+class ExclamationCircleIconPoint extends React.Component<any> {
```

Comment:
Is it possible to convert this to a functional component? :)

Reply:
@brumik It's a class on purpose to be able to replicate the behaviour from https://formidable.com/open-source/victory/docs/common-props/#datacomponent

Reply:
@ZitaNemeckova This works:

```ts
export const dataComponentMapper = (name: string): React.ReactNode => {
  switch (name) {
    case 'ExclamationCircleIcon':
      return <ExclamationCircleIconPoint />;
    default:
      return null;
  }
};

interface Props {
  x?: number;
  y?: number;
}

const ExclamationCircleIconPoint: FC<Props> = ({ x, y, ...props }) => {
  // TODO Where the -8 came from in this?
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  return <ExclamationCircleIcon x={x-8} y={y-8} {...props} />;
};

```

Reply:
So we can have the dataComponent an object of `Record<string, React.Component>` and do the mapping like in the `onClick` method. The mapper could apply the `-8` padding on x/y